src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -140,16 +141,26 @@ protected override Payload ConstructHeadersPayload(GetHeadersPayload getHeadersP
                         this.logger.LogTrace("(-)[NO_PH_AVAILABLE]");
                         break;
                     }
+
+                    // We need to check that the headers we are returning are consecutive.
+                    // If a specific header is found not to follow on from the previous one,
+                    // we just return what we have that are consecutive.
+                    var previous = lastHeader ?? fork;
+                    if (provenBlockHeader.HashPrevBlock != previous.HashBlock)
```

Comment:
```
if (provenBlockHeader.HashPrevBlock != header.Previous.HashBlock)
```

instead of this and prev line would probably be better

Reply:
We should compare against the value that was returned form the PH store right?

Reply:
That line and what is currently in the code does it, yes

Reply:
Does something need to change here then? (Confused.com)